### PR TITLE
[Docs] Add training and testing tricks

### PR DESCRIPTION
**Pull request by hhaAndroid**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation for this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit tests to ensure the correctness.
3. If the modification has a potential influence on downstream projects, this PR should be tested with downstream projects, like MMDetection or MMClassification.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: docs/zh_cn/recommended_topics/training_testing_tricks.md [hunk 1]
```diff
@@ -0,0 +1,308 @@
...
+<img src="https://user-images.githubusercontent.com/17425982/224928241-89dac006-392b-445d-87e8-a9e268825401.png" width="1000"/>
+</div>
+
+图片来自 [detectron2 issue](https://github.com/facebookresearch/detectron2/issues/1909)。黄色打叉的区域表示 `iscrowd` 标注。原因有两个方便：
+
+- 这个区域不是真的人，例如海报上的人
+- 该区域过于拥挤，很难标注
+
+在该场景下，你不能简单的将这类标注删掉，因为你一旦删掉就表示当做背景区域来训练了，但是其和背景是不一样的，首先海报上的人和真人很像，并且拥挤区域确实有人只是不好标注。如果你简单的将其当做背景训练，那么会造成漏报。最合适的做法应该是当做忽略区域即该区域的任何输出都
+直接忽略，不计算任何 Loss，不强迫模型拟合。
```
Comment (PeterH0323) [inline docs/zh_cn/recommended_topics/training_testing_tricks.md:99]: 这里可以不需要回车换行

File: docs/zh_cn/recommended_topics/training_testing_tricks.md [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+
+MMYOLO 在 YOLOv5 上简单快速的验证了 `iscrowd` 标注的作用，性能如下所示：
+
+| Backbone | ignore_iof_thr | box AP50(CrowDHuman Metric) |  MR  |  JI   |
+| :------: | :------------: | :-------------------------: | :--: | :---: |
+| YOLOv5-s |       -1       |            85.79            | 48.7 | 75.33 |
+| YOLOv5-s |      0.5       |            86.17            | 48.8 | 75.87 |
+
+`ignore_iof_thr`为 -1 表示不考虑忽略标签，可以看出性能有一定程度的提升，具体见 [CrowdHuman 结果](https://github.com/open-mmlab/mmyolo/blob/main/configs/yolov5/README.md#crowdhuman)。 如果你的自定义数据集上也有上述情况，则建议你考虑 ignore 场景避免不确定性标注。
```
Comment (PeterH0323) [inline docs/zh_cn/recommended_topics/training_testing_tricks.md:108]:
```suggestion
`ignore_iof_thr`为 `-1` 表示不考虑忽略标签，可以看出性能有一定程度的提升，具体见 [CrowdHuman 结果](https://github.com/open-mmlab/mmyolo/blob/main/configs/yolov5/README.md#crowdhuman)。 如果你的自定义数据集上也有上述情况，则建议你考虑 ignore 场景避免不确定性标注。
```

File: docs/zh_cn/recommended_topics/training_testing_tricks.md [hunk 3]
```diff
@@ -0,0 +1,308 @@
...
+
+#### 7 使用知识蒸馏
+
+知识蒸馏是一个被广泛使用的技巧，可以将大模型性能转移到小模型上从而提升小模型检测性能。 目前 MMYOLO 和 MMRazor 联合支持了该功能，并在 RTMDet 上进行了初步验证。
```
Comment (PeterH0323) [inline docs/zh_cn/recommended_topics/training_testing_tricks.md:112]:
```suggestion
知识蒸馏是一个被广泛使用的技巧，可以将大模型性能转移到小模型上从而提升小模型检测性能。 目前 MMYOLO 联合 MMRazor 以支持该功能，并在 RTMDet 上进行了初步验证。
```

File: docs/zh_cn/recommended_topics/training_testing_tricks.md [hunk 4]
```diff
@@ -0,0 +1,308 @@
...
+    return_iou=True),
+loss_obj=dict(
+    type='mmdet.CrossEntropyLoss',
+    use_sigmoid=True,
+    reduction='mean',
+    loss_weight=loss_obj_weight *
+    ((img_scale[0] / 640)**2 * 3 / num_det_layers)),
+```
+
+loss_cls 可以根据自定义类别数和检测层数对 loss_weight 进行自适应缩放，loss_bbox 可以根据检测层数进行自适应计算，而 loss_obj 可以根据输入图片大小和检测层数进行自适应缩放。这种策略可以让用户不用去设置 Loss 权重超参。
```
Comment (PeterH0323) [inline docs/zh_cn/recommended_topics/training_testing_tricks.md:197]:
```suggestion
`loss_cls` 可以根据自定义类别数和检测层数对 `loss_weight` 进行自适应缩放，`loss_bbox` 可以根据检测层数进行自适应计算，而 `loss_obj` 可以根据输入图片大小和检测层数进行自适应缩放。这种策略可以让用户不用去设置 Loss 权重超参。
```

File: docs/zh_cn/recommended_topics/training_testing_tricks.md [hunk 5]
```diff
@@ -0,0 +1,308 @@
...
+#### 1 推理时避免一个检测框输出多个类别
+
+YOLOv5 在训练分类分支时候采用的是 BCE Loss 即 `use_sigmoid=True`。假设物体类别数是 4，那么分类分支输出的类别数是 4 而不是 5，并且由于使用的是 sigmoid 而非 softmax 预测模式，很可能在某个预测位置出
+出现多个满足过滤阈值的检测框，也就是会出现一个预测 bbox 对应了多个预测 label。如下图所示
+
+<div align=center>
+<img alt="multi-lable" src="https://user-images.githubusercontent.com/17425982/226272053-75b2bd2d-99ad-4d47-9a24-a20c742fd323.png" width=800 />
+</div>
+
+一般在计算 mAP 时候过滤阈值为 0.001，由于 sigmoid 非竞争性预测模式会导致一个框对应多个 label。这种计算方式可以提高 mAP 计算时候的召回率，但是实际落地应用会不方便。
```
Comment (PeterH0323) [inline docs/zh_cn/recommended_topics/training_testing_tricks.md:254]:
```suggestion
一般在计算 mAP 时候过滤阈值为 `0.001`，由于 sigmoid 非竞争性预测模式会导致一个框对应多个 label。这种计算方式可以提高 mAP 计算时候的召回率，但是实际落地应用会不方便。
```

File: docs/zh_cn/recommended_topics/training_testing_tricks.md [hunk 6]
```diff
@@ -0,0 +1,303 @@
...
+
+## 训练技巧
+
+### 提升检测性能
+
+#### 1 开启多尺度训练
+
+在目标检测领域，多尺度训练是一个非常常用的技巧，但是在 YOLO 中大部分模型的训练输入都是单尺度的 640x640，原因有两个方面：
+
+1. 单尺度训练效果高。当训练 epoch 在 300 或者 500 时候训练效率是用户非常关注的，多尺度训练会比较慢
```
Comment (RangeKing) [inline docs/zh_cn/recommended_topics/training_testing_tricks.md:13]:
```suggestion
1. 单尺度训练速度快。当训练 epoch 在 300 或者 500 的时候训练效率是用户非常关注的，多尺度训练会比较慢
```

File: docs/zh_cn/recommended_topics/training_testing_tricks.md [hunk 7]
```diff
@@ -0,0 +1,303 @@
...
+2. 训练 pipeline 中隐含了多尺度增强，等价于应用了多尺度训练，典型的如 `Mosaic`、`RandomAffine` 和 `Resize` 等，故没有必要再次引入模型输入的多尺度训练
+
+在 COCO 数据集上进行了简单实验，如果直接在 YOLOv5 的 DataLoader 输出后再次引入多尺度训练增强实际性能提升非常小，但是这不代表用户自定义数据集微调模式下没有明显增益。如果想在 MMYOLO 中对 YOLO 系列算法开启多尺度训练，可以参考 [多尺度训练文档](../common_usage/ms_training_testing.md)
+
+#### 2 使用 mask 标注优化目标检测性能
```
Comment (RangeKing) [inline docs/zh_cn/recommended_topics/training_testing_tricks.md:18]:
```suggestion
#### 2 使用 Mask 标注优化目标检测性能
```

File: docs/zh_cn/recommended_topics/training_testing_tricks.md [hunk 8]
```diff
@@ -0,0 +1,303 @@
...
+<div align="center">
+<img src="https://user-images.githubusercontent.com/17425982/224928241-89dac006-392b-445d-87e8-a9e268825401.png" width="1000"/>
+</div>
+
+图片来自 [detectron2 issue](https://github.com/facebookresearch/detectron2/issues/1909)。黄色打叉的区域表示 `iscrowd` 标注。原因有两个方便：
+
+- 这个区域不是真的人，例如海报上的人
+- 该区域过于拥挤，很难标注
+
+在该场景下，你不能简单的将这类标注删掉，因为你一旦删掉就表示当做背景区域来训练了，但是其和背景是不一样的，首先海报上的人和真人很像，并且拥挤区域确实有人只是不好标注。如果你简单的将其当做背景训练，那么会造成漏报。最合适的做法应该是当做忽略区域即该区域的任何输出都直接忽略，不计算任何 Loss，不强迫模型拟合。
```
Comment (RangeKing) [inline docs/zh_cn/recommended_topics/training_testing_tricks.md:96]:
```suggestion
在该场景下，你不能简单的将这类标注删掉，因为你一旦删掉就表示当做背景区域来训练了，但是其和背景是不一样的，首先海报上的人和真人很像，并且拥挤区域确实有人只是不好标注。如果你简单的将其当做背景训练，那么会造成漏报。最合适的做法应该是把拥挤区域当做忽略区域即该区域的任何输出都直接忽略，不计算任何 Loss，不强迫模型拟合。
```

File: docs/zh_cn/recommended_topics/training_testing_tricks.md [hunk 9]
```diff
@@ -0,0 +1,303 @@
...
+    use_sigmoid=True,
+    reduction='mean',
+    loss_weight=loss_obj_weight *
+    ((img_scale[0] / 640)**2 * 3 / num_det_layers)),
+```
+
+`loss_cls` 可以根据自定义类别数和检测层数对 `loss_weight` 进行自适应缩放，`loss_bbox` 可以根据检测层数进行自适应计算，而 `loss_obj` 可以根据输入图片大小和检测层数进行自适应缩放。这种策略可以让用户不用去设置 Loss 权重超参。
+需要说明的是：这个只是经验规则，并不是说是最佳设置组合，只是作为一个参考。
+
+#### 2 Weight Decay 和 Loss 输出值基于 Batch Size 自适应， 少 2 个超参
```
Comment (RangeKing) [inline docs/zh_cn/recommended_topics/training_testing_tricks.md:196]:
```suggestion
#### 2 Weight Decay 和 Loss 输出值基于 Batch Size 自适应，少 2 个超参
```
